### DISCUSSION TITLE
App router - No "router.events" - How can I save to session storage on route change?

### DISCUSSION THREAD

**Discussion by damyco** [vercel/next.js#64971] [Help]
### Summary

I have a use case where I need to save to session storage on route change, before, in Pages router I had something like that:
```
    useEffect(() => {
        const handleRouteChange = () => {
             setSessionStorage(); 
        };

        router.events.on('beforeHistoryChange', handleRouteChange);

        return () => {
            router.events.off('beforeHistoryChange', handleRouteChange);
        };
    }, []);
```

How can I achieve the same functionality using App router? How can I call my setSessionStorage() function if I navigate away from the page?

**Comment by icyJoseph** [comment]
If you have a component that only renders on a given page, you could use the clean up function of an effect, but that'd have some StrictMode trickery going on.

Perhaps you could adapt this one: https://nextjs.org/docs/app/api-reference/functions/use-router#router-events ?
